core-rust/state-manager/src/jni/node_rust_environment.rs

```diff
@@ -122,13 +125,16 @@ pub struct JNINodeRustEnvironment {
     pub state_manager: StateManager,
     pub metric_registry: Arc<Registry>,
     pub running_task_tracker: UntilDropTracker,
+    pub node_store: Arc<DbLock<ActualNodeDatabase>>,
```

Comment:
Three things:
* I thought we were having _separate_ databases for Safety and Address Book (and possibly migrations)?
* If we did have one database, in my opinion we shouldn't call it the "node database" - it's a very misleading name as it doesn't include e.g. the ledger database.
* I don't see why we need a lock around them here - this seems like it will be uselessly contentioned (particularly with a combined database, where e.g. a write of the address book would block reading the safety store) -  Rocks does locking internally anyway where it needs to.

Reply:
Perhaps this is a misunderstanding because I got the conclusion that these stores should be separate from the ledger, but otherwise it's fine to keep them together. But I see no problems separating them now.